From 347c13c78688f53e07315d73476e7d7cd1ad58d5 Mon Sep 17 00:00:00 2001
From: Vml11 <victorbcn98@gmail.com>
Date: Fri, 6 Apr 2018 02:40:27 +0200
Subject: [PATCH] Added-TNTDispenseEvent


diff --git a/src/main/java/me/torciv/fluxspigot/event/entity/TNTDispensedEvent.java b/src/main/java/me/torciv/fluxspigot/event/entity/TNTDispensedEvent.java
new file mode 100644
index 00000000..5a20a78f
--- /dev/null
+++ b/src/main/java/me/torciv/fluxspigot/event/entity/TNTDispensedEvent.java
@@ -0,0 +1,44 @@
+package me.torciv.fluxspigot.event.entity;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.EntityEvent;
+
+/**
+ * Fired whenever a TNT is Dispensed
+ */
+public class TNTDispensedEvent extends EntityEvent implements Cancellable {
+
+    private boolean cancelled;
+    private static final HandlerList handlerList = new HandlerList();
+
+    public TNTDispensedEvent(Entity tnt) {
+        super(tnt);
+        this.cancelled = false;
+    }
+
+    // Cancellable
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancel) {
+        this.cancelled = cancel;
+    }
+
+    // handlers
+    
+    @Override
+    public HandlerList getHandlers() {
+        return handlerList;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlerList;
+    }
+
+}
-- 
2.13.0.windows.1

