From 6b8a9571fac4851146b55240a44c0d9ce3031996 Mon Sep 17 00:00:00 2001
From: Vml11 <victorbcn98@gmail.com>
Date: Thu, 5 Apr 2018 23:34:29 +0200
Subject: [PATCH] Started project. Push to see if fork actually works on github


diff --git a/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java b/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java
new file mode 100644
index 00000000..8092a372
--- /dev/null
+++ b/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java
@@ -0,0 +1,9 @@
+package me.torciv.fluxspigot;
+
+import java.io.File;
+
+public class FluxSpigotConfig {
+
+    private static File CONFIG_FILE;
+    
+}
-- 
2.13.0.windows.1

