From 95f5cdf62bcedb3a65c8216d02ac938e700f336e Mon Sep 17 00:00:00 2001
From: Vml11 <victorbcn98@gmail.com>
Date: Thu, 5 Apr 2018 23:40:46 +0200
Subject: [PATCH] Started project. Push to see if fork actually works on github


diff --git a/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java b/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java
index 8092a372..12e25b94 100644
--- a/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java
+++ b/src/main/java/me/torciv/fluxspigot/FluxSpigotConfig.java
@@ -5,5 +5,5 @@ import java.io.File;
 public class FluxSpigotConfig {
 
     private static File CONFIG_FILE;
-    
+
 }
-- 
2.13.0.windows.1

