From b8a9162cee3bad5d99741b055fba4fcb4dc7e9e4 Mon Sep 17 00:00:00 2001
From: Vml11 <victorbcn98@gmail.com>
Date: Fri, 6 Apr 2018 18:42:52 +0200
Subject: [PATCH] Added-left-shooting-control


diff --git a/src/main/java/me/torciv/fluxspigot/FluxSpigotWorldConfig.java b/src/main/java/me/torciv/fluxspigot/FluxSpigotWorldConfig.java
index 72a3f9df4..f00916844 100644
--- a/src/main/java/me/torciv/fluxspigot/FluxSpigotWorldConfig.java
+++ b/src/main/java/me/torciv/fluxspigot/FluxSpigotWorldConfig.java
@@ -79,10 +79,16 @@ public class FluxSpigotWorldConfig {
 
     public boolean fixLeftShooting;
     private void fixLeftShooting(){
-        fixLeftShooting = getBoolean("cannons.fix-left-shooting-cannoning", false);
+        fixLeftShooting = getBoolean("cannons.fix-left-shooting-cannoning", true);
         if(fixLeftShooting) log("fixing left-Shooting cannoning");
     }
 
+    public double leftShootingAcurancy;
+    private void leftShootingAcurancy(){
+        leftShootingAcurancy = getDouble("cannons.left-shooting-accurancy", 120.0);
+        if(fixLeftShooting) log("leftShootingAcurancy set to " + leftShootingAcurancy);
+    }
+
     public boolean fireTNTDispenseEvent;
     private void fireTNTDispenseEvent(){
         fireTNTDispenseEvent = getBoolean("cannons.fire-tnt-dispense-event", true);
diff --git a/src/main/java/me/torciv/fluxspigot/utils/Util.java b/src/main/java/me/torciv/fluxspigot/utils/Util.java
new file mode 100644
index 000000000..b8604bc7d
--- /dev/null
+++ b/src/main/java/me/torciv/fluxspigot/utils/Util.java
@@ -0,0 +1,46 @@
+package me.torciv.fluxspigot.utils;
+
+import net.minecraft.server.EntityFallingBlock;
+import org.bukkit.Location;
+import org.bukkit.entity.FallingBlock;
+
+public class Util {
+
+
+    public static double fallingBlockTravelDistance(FallingBlock entity){
+        double distance = -1.0;
+        double from [] = new double [2];
+        double to [] = new double [2];
+
+        Location spawn = entity.getSourceLoc();
+        from[0] = spawn.getX();
+        from[1] = spawn.getZ();
+
+        Location explosion = entity.getLocation();
+        to[0] = explosion.getX();
+        to[1] = explosion.getZ();
+        distance = calculateDistance(from,to);
+        
+        return distance;
+    }
+
+    public static double fallingBlockTravelDistances(EntityFallingBlock entity){
+        double distance = -1.0;
+        double from [] = new double [2];
+        double to [] = new double [2];
+
+        Location spawn = entity.sourceLoc;
+        from[0] = spawn.getX();
+        from[1] = spawn.getZ();
+
+        to[0] = entity.locX;
+        to[1] = entity.locZ;
+        distance = calculateDistance(from,to);
+
+        return distance;
+    }
+
+    private static double calculateDistance(double[] from, double [] to){
+        return Math.sqrt(((Math.pow((to[0]-from[0]),2) + Math.pow((to[1]-from[1]),2))));
+    }
+}
diff --git a/src/main/java/net/minecraft/server/EntityFallingBlock.java b/src/main/java/net/minecraft/server/EntityFallingBlock.java
index e3bfd8683..c8600dbd5 100644
--- a/src/main/java/net/minecraft/server/EntityFallingBlock.java
+++ b/src/main/java/net/minecraft/server/EntityFallingBlock.java
@@ -18,6 +18,8 @@ public class EntityFallingBlock extends Entity {
     public NBTTagCompound tileEntityData;
     public org.bukkit.Location sourceLoc; // PaperSpigot
 
+    public boolean traveledTooFar = false;
+
     // PaperSpigot start - Add FallingBlock source location API
     public EntityFallingBlock(World world) {
         this(null, world);
@@ -83,6 +85,7 @@ public class EntityFallingBlock extends Entity {
             }
 
             this.motY -= 0.03999999910593033D;
+            System.out.println("x " + motX + " y " + motY + " z " + motZ);
             this.move(this.motX, this.motY, this.motZ);
 
             // PaperSpigot start - Remove entities in unloaded chunks
diff --git a/src/main/java/net/minecraft/server/Explosion.java b/src/main/java/net/minecraft/server/Explosion.java
index f4ccd8d4d..ba0c49bb3 100644
--- a/src/main/java/net/minecraft/server/Explosion.java
+++ b/src/main/java/net/minecraft/server/Explosion.java
@@ -10,10 +10,16 @@ import java.util.Map;
 import java.util.Random;
 
 // CraftBukkit start
+import me.torciv.fluxspigot.utils.Util;
+import org.bukkit.craftbukkit.entity.CraftEntity;
 import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.entity.FallingBlock;
+import org.bukkit.entity.TNTPrimed;
 import org.bukkit.event.entity.EntityExplodeEvent;
 import org.bukkit.Location;
 import org.bukkit.event.block.BlockExplodeEvent;
+import org.bukkit.metadata.FixedMetadataValue;
+import org.bukkit.util.Vector;
 // CraftBukkit end
 
 public class Explosion {
@@ -147,6 +153,23 @@ public class Explosion {
                         // CraftBukkit end
                         double d14 = entity instanceof EntityHuman && world.paperSpigotConfig.disableExplosionKnockback ? 0 : EnchantmentProtection.a(entity, d13); // PaperSpigot
 
+                        if(entity instanceof  EntityFallingBlock){
+                            EntityFallingBlock entityFallingBlock = (EntityFallingBlock) entity;
+                            CraftEntity centitiy = CraftEntity.getEntity(world.getServer(),entity);
+                            if(!entityFallingBlock.traveledTooFar){
+                                double traveled = Util.fallingBlockTravelDistances(entityFallingBlock);
+                                System.out.println(traveled);
+                                if(traveled >= 20.0){
+                                    entityFallingBlock.traveledTooFar = true;
+                                    entity.motX = 0;
+                                    entity.motZ = 0;
+                                }
+                            }else{
+                                entity.motX = 0;
+                                entity.motZ = 0;
+
+                            }
+                        }
                         // PaperSpigot start - Fix cannons
                         /*
                         entity.motX += d8 * d14;
@@ -198,6 +221,7 @@ public class Explosion {
 
             boolean cancelled;
             List<org.bukkit.block.Block> bukkitBlocks;
+            List<org.bukkit.entity.Entity> entities = Lists.newArrayList();
             float yield;
 
             if (explode != null) {
@@ -206,6 +230,15 @@ public class Explosion {
                 cancelled = event.isCancelled();
                 bukkitBlocks = event.blockList();
                 yield = event.getYield();
+
+                // FluxSpigot Handle Leftshoot get entities nearby
+                if(world.fluxSpigotConfig.fixLeftShooting){
+                    if(explode instanceof TNTPrimed){
+                        entities = (List<org.bukkit.entity.Entity>)location.getWorld().getNearbyEntities(location, volume, volume, volume);
+                    }
+                }
+                // End FluxSpigot Handle LeftShoot entities
+
             } else {
                 BlockExplodeEvent event = new BlockExplodeEvent(location.getBlock(), blockList, 0.3F);
                 this.world.getServer().getPluginManager().callEvent(event);
@@ -225,6 +258,26 @@ public class Explosion {
                 this.wasCanceled = true;
                 return;
             }
+
+            // FluxSpigot Handle LeftShoot
+
+            if(world.fluxSpigotConfig.fixLeftShooting){
+                Iterator fbiterator = entities.iterator();
+                while(fbiterator.hasNext()){
+                    org.bukkit.entity.Entity fb = (org.bukkit.entity.Entity) fbiterator.next();
+                    if(fb instanceof FallingBlock){
+                        FallingBlock fallingBlock = (FallingBlock) fb;
+                        double traveled = Util.fallingBlockTravelDistance(fallingBlock);
+                        if(traveled != -1 && traveled >= world.fluxSpigotConfig.leftShootingAcurancy ){
+                            continue;
+                            //double y_vel = fb.getVelocity().getY();
+                            //fb.setVelocity(new Vector(0.0,y_vel,0.0));
+                        }
+                    }
+                }
+            }
+            // End FluxSpigot handle leftshoot
+
             // CraftBukkit end
             iterator = this.blocks.iterator();
 
diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
index db915033b..25391ccfd 100644
--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
@@ -1,16 +1,9 @@
 package org.bukkit.craftbukkit.event;
 
-import java.net.InetAddress;
-import java.util.ArrayList;
-import java.util.EnumMap;
-import java.util.List;
-import java.util.Map;
-
 import com.google.common.base.Function;
 import com.google.common.base.Functions;
-
 import net.minecraft.server.*;
-
+import net.minecraft.server.Entity;
 import org.bukkit.Bukkit;
 import org.bukkit.Material;
 import org.bukkit.Server;
@@ -32,19 +25,7 @@ import org.bukkit.craftbukkit.inventory.CraftItemStack;
 import org.bukkit.craftbukkit.inventory.CraftMetaBook;
 import org.bukkit.craftbukkit.util.CraftDamageSource;
 import org.bukkit.craftbukkit.util.CraftMagicNumbers;
-import org.bukkit.entity.Arrow;
-import org.bukkit.entity.Creeper;
-import org.bukkit.entity.EntityType;
-import org.bukkit.entity.Firework;
-import org.bukkit.entity.Horse;
-import org.bukkit.entity.LightningStrike;
-import org.bukkit.entity.LivingEntity;
-import org.bukkit.entity.Pig;
-import org.bukkit.entity.PigZombie;
-import org.bukkit.entity.Player;
-import org.bukkit.entity.Projectile;
-import org.bukkit.entity.ThrownExpBottle;
-import org.bukkit.entity.ThrownPotion;
+import org.bukkit.entity.*;
 import org.bukkit.event.Cancellable;
 import org.bukkit.event.Event;
 import org.bukkit.event.block.*;
@@ -61,6 +42,12 @@ import org.bukkit.event.server.ServerListPingEvent;
 import org.bukkit.inventory.InventoryView;
 import org.bukkit.inventory.meta.BookMeta;
 
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
 public class CraftEventFactory {
     public static final DamageSource MELTING = CraftDamageSource.copyOf(DamageSource.BURN);
     public static final DamageSource POISON = CraftDamageSource.copyOf(DamageSource.MAGIC);
-- 
2.13.0.windows.1

