From f85adc3c78f51db8dc12c0685740c84267a2965f Mon Sep 17 00:00:00 2001
From: daniel <cens097541@gmail.com>
Date: Sat, 7 Apr 2018 15:21:11 +0200
Subject: [PATCH] flux-optimize-tnt


diff --git a/pom.xml b/pom.xml
index d116fa455..d96aa102a 100644
--- a/pom.xml
+++ b/pom.xml
@@ -123,7 +123,7 @@
             <artifactId>gitdescribe-maven-plugin</artifactId>
             <version>1.3</version>
             <configuration>
-                <outputPrefix>git-TacoSpigot-</outputPrefix>
+                <outputPrefix>FluxSpigot-</outputPrefix>
                 <scmDirectory>..</scmDirectory>
             </configuration>
             <executions>
diff --git a/src/main/java/net/minecraft/server/EntityTNTPrimed.java b/src/main/java/net/minecraft/server/EntityTNTPrimed.java
index 872d9659c..dee1cfa3c 100644
--- a/src/main/java/net/minecraft/server/EntityTNTPrimed.java
+++ b/src/main/java/net/minecraft/server/EntityTNTPrimed.java
@@ -27,17 +27,22 @@ public class EntityTNTPrimed extends Entity {
     public EntityTNTPrimed(org.bukkit.Location loc, World world, double d0, double d1, double d2, EntityLiving entityliving) {
         this(loc, world);
         this.setPosition(d0, d1, d2);
-        float f = (float) (Math.random() * 3.1415927410125732D * 2.0D);
+        /// fluxspigot optimize tnt /////
+        if (!world.paperSpigotConfig.fixCannons){
+            float f = (float) (Math.random() * 3.1415927410125732D * 2.0D);
+            this.motX = (double) (-((float) Math.sin((double) f)) * 0.02F);
+            this.motZ = (double) (-((float) Math.cos((double) f)) * 0.02F);
+        }
+        this.motX = this.motZ = 0.0F;
+        /// fluxspigot optimize tnt /////
 
-        this.motX = (double) (-((float) Math.sin((double) f)) * 0.02F);
         this.motY = 0.20000000298023224D;
-        this.motZ = (double) (-((float) Math.cos((double) f)) * 0.02F);
         this.fuseTicks = 80;
         this.lastX = d0;
         this.lastY = d1;
         this.lastZ = d2;
         this.source = entityliving;
-        if (world.paperSpigotConfig.fixCannons) this.motX = this.motZ = 0.0F; // PaperSpigot - Fix cannons
+        // PaperSpigot - Fix cannons
     }
 
     protected void h() {}
@@ -90,7 +95,8 @@ public class EntityTNTPrimed extends Entity {
             // CraftBukkit end
         } else {
             this.W();
-            this.world.addParticle(EnumParticle.SMOKE_NORMAL, this.locX, this.locY + 0.5D, this.locZ, 0.0D, 0.0D, 0.0D, new int[0]);
+            /// fluxspigot optimize tnt /////
+            //this.world.addParticle(EnumParticle.SMOKE_NORMAL, this.locX, this.locY + 0.5D, this.locZ, 0.0D, 0.0D, 0.0D, new int[0]);
         }
 
     }
@@ -191,6 +197,9 @@ public class EntityTNTPrimed extends Entity {
         this.motY = oldMotY;
         this.motZ = oldMotZ;
 
+        ////// fluxspigot optimice tnt //////
+        // remove unnecesary cosmetics reducing client and server operations//
+        /*
         if (this.inWater) {
             // Send position and velocity updates to nearby players on every tick while the TNT is in water.
             // This does pretty well at keeping their clients in sync with the server.
@@ -207,6 +216,7 @@ public class EntityTNTPrimed extends Entity {
                 }
             }
         }
+        */
 
         return this.inWater;
     }
diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index 1fc2388ea..41b593c5e 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -522,9 +522,17 @@ public class EntityTrackerEntry {
                 return packetplayoutspawnentity;
             } else if (this.tracker instanceof EntityEgg) {
                 return new PacketPlayOutSpawnEntity(this.tracker, 62);
-            } else if (this.tracker instanceof EntityTNTPrimed) {
+            }
+
+            //fluxspigot invisible tnt //
+            /*
+            else if (this.tracker instanceof EntityTNTPrimed) {
                 return new PacketPlayOutSpawnEntity(this.tracker, 50);
-            } else if (this.tracker instanceof EntityEnderCrystal) {
+            }
+           */
+            // fluxspigot invisible tnt //
+
+            else if (this.tracker instanceof EntityEnderCrystal) {
                 return new PacketPlayOutSpawnEntity(this.tracker, 51);
             } else if (this.tracker instanceof EntityFallingBlock) {
                 EntityFallingBlock entityfallingblock = (EntityFallingBlock) this.tracker;
diff --git a/src/main/java/net/minecraft/server/Explosion.java b/src/main/java/net/minecraft/server/Explosion.java
index b9375223a..b826c35c6 100644
--- a/src/main/java/net/minecraft/server/Explosion.java
+++ b/src/main/java/net/minecraft/server/Explosion.java
@@ -199,12 +199,16 @@ public class Explosion {
         float volume = source instanceof EntityTNTPrimed ? world.paperSpigotConfig.tntExplosionVolume : 4.0F;
         this.world.makeSound(this.posX, this.posY, this.posZ, "random.explode", volume, (1.0F + (this.world.random.nextFloat() - this.world.random.nextFloat()) * 0.2F) * 0.7F);
         // PaperSpigot end
+
+        // fluxspigot optimice tnt //
+        // remove unncesary particles and cosmetics //
+        /*
         if (this.size >= 2.0F && this.b) {
             this.world.addParticle(EnumParticle.EXPLOSION_HUGE, this.posX, this.posY, this.posZ, 1.0D, 0.0D, 0.0D, new int[0]);
         } else {
             this.world.addParticle(EnumParticle.EXPLOSION_LARGE, this.posX, this.posY, this.posZ, 1.0D, 0.0D, 0.0D, new int[0]);
         }
-
+        */
         Iterator iterator;
         BlockPosition blockposition;
 
@@ -280,8 +284,13 @@ public class Explosion {
                     d3 *= d7;
                     d4 *= d7;
                     d5 *= d7;
+
+                    //fluxspigot optimice tnt //
+                    // remove unncesary cosmetics madafaka //
+                    /*
                     this.world.addParticle(EnumParticle.EXPLOSION_NORMAL, (d0 + this.posX * 1.0D) / 2.0D, (d1 + this.posY * 1.0D) / 2.0D, (d2 + this.posZ * 1.0D) / 2.0D, d3, d4, d5, new int[0]);
                     this.world.addParticle(EnumParticle.SMOKE_NORMAL, d0, d1, d2, d3, d4, d5, new int[0]);
+                    */
                 }
 
                 if (block.getMaterial() != Material.AIR) {
diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 40a066ca5..79db9e4a8 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -1067,15 +1067,20 @@ public class WorldServer extends World implements IAsyncTaskHandler {
             explosion.clearBlocks();
         }
 
-        Iterator iterator = this.players.iterator();
+        // fluxspigot fix tnt lag //
 
+        /*
+        Iterator iterator = this.players.iterator();
         while (iterator.hasNext()) {
             EntityHuman entityhuman = (EntityHuman) iterator.next();
 
             if (entityhuman.e(d0, d1, d2) < 4096.0D) {
                 ((EntityPlayer) entityhuman).playerConnection.sendPacket(new PacketPlayOutExplosion(d0, d1, d2, f, explosion.getBlocks(), (Vec3D) explosion.b().get(entityhuman)));
             }
+
         }
+        */
+        // fluxspigot fix tnt lag //
 
         return explosion;
     }
-- 
2.14.3.windows.1

