From 9d6f1dc7f55491ceb51f1a75b89ff0bc55be7dba Mon Sep 17 00:00:00 2001
From: Vml11 <victorbcn98@gmail.com>
Date: Sat, 7 Apr 2018 15:39:38 +0200
Subject: [PATCH] Added-left-shooting-control


diff --git a/src/main/java/net/minecraft/server/EntityFallingBlock.java b/src/main/java/net/minecraft/server/EntityFallingBlock.java
index 7b976e6b5..509db3f19 100644
--- a/src/main/java/net/minecraft/server/EntityFallingBlock.java
+++ b/src/main/java/net/minecraft/server/EntityFallingBlock.java
@@ -83,9 +83,25 @@ public class EntityFallingBlock extends Entity {
                     return;
                 }
             }
-
+            System.out.println("X1: " + this.motX + " Y1: " + this.motY + " Z1: " + this.motZ);
             this.motY -= 0.03999999910593033D;
-            System.out.println("x " + motX + " y " + motY + " z " + motZ);
+
+            // Handle leftshooting FluxSpigot
+            double defaultMotX = 0.9800000190734863D;
+            double defaultMotZ = 0.9800000190734863D;
+
+
+            if(this.traveledTooFar){
+                defaultMotX = 0.0D;
+                defaultMotZ = 0.0D;
+                this.motX = 0.0D;
+                this.motZ = 0.0D;
+                System.out.println("entrado");
+            }
+
+            System.out.println("X2: " + this.motX + " Y2: " + this.motY + " Z2: " + this.motZ);
+            // Handle LeftShooting FluxSpigot
+
             this.move(this.motX, this.motY, this.motZ);
 
             // PaperSpigot start - Remove entities in unloaded chunks
@@ -104,14 +120,10 @@ public class EntityFallingBlock extends Entity {
             }
             // PaperSpigot end
 
-            this.motX *= 0.9800000190734863D;
+            this.motX *= defaultMotX;
             this.motY *= 0.9800000190734863D;
-            this.motZ *= 0.9800000190734863D;
+            this.motZ *= defaultMotZ;
 
-            if(this.traveledTooFar){
-                this.motZ = 0.0D;
-                this.motX = 0.0D;
-            }
 
             if (!this.world.isClientSide) {
                 blockposition = new BlockPosition(this);
diff --git a/src/main/java/net/minecraft/server/Explosion.java b/src/main/java/net/minecraft/server/Explosion.java
index ba0c49bb3..e95cf43b9 100644
--- a/src/main/java/net/minecraft/server/Explosion.java
+++ b/src/main/java/net/minecraft/server/Explosion.java
@@ -160,15 +160,15 @@ public class Explosion {
                                 double traveled = Util.fallingBlockTravelDistances(entityFallingBlock);
                                 System.out.println(traveled);
                                 if(traveled >= 20.0){
+                                    entityFallingBlock.motX = 0.0D;
+                                    entityFallingBlock.motZ = 0.0D;
                                     entityFallingBlock.traveledTooFar = true;
-                                    entity.motX = 0;
-                                    entity.motZ = 0;
                                 }
                             }else{
-                                entity.motX = 0;
-                                entity.motZ = 0;
-
+                                entityFallingBlock.motX = 0.0D;
+                                entityFallingBlock.motZ = 0.0D;
                             }
+
                         }
                         // PaperSpigot start - Fix cannons
                         /*
@@ -177,7 +177,9 @@ public class Explosion {
                         entity.motZ += d10 * d14;
                         */
                         // This impulse method sets the dirty flag, so clients will get an immediate velocity update
+
                         entity.g(d8 * d14, d9 * d14, d10 * d14);
+
                         // PaperSpigot end
 
                         if (entity instanceof EntityHuman && !((EntityHuman) entity).abilities.isInvulnerable && !world.paperSpigotConfig.disableExplosionKnockback) { // PaperSpigot
-- 
2.13.0.windows.1

